Also watch for flake8 errors (you can view them in the PYLINT results of the travis build :
```
partner_compassion/models/partner_compassion.py:76:18: E111 indentation is not a multiple of four
partner_compassion/models/partner_compassion.py:77:18: E111 indentation is not a multiple of four
```

--- a/partner_compassion/models/partner_compassion.py
+++ b/partner_compassion/models/partner_compassion.py
@@ -79,14 +79,13 @@ class ResPartner(models.Model):
     partner_duplicate_ids = fields.Many2many(
         'res.partner', 'res_partner_duplicates', 'partner_id',
         'duplicate_id', readonly=True)
-    church_member_count = fields.Integer(compute='_is_church',
-                                         readonly=True)
+    church_member_count = fields.Integer(compute='_is_church')
 
     ##########################################################################
     #                             FIELDS METHODS                             #
     ##########################################################################
     @api.multi
-    @api.depends('category_id', 'member_ids', 'church_member_count')
+    @api.depends('category_id', 'member_ids')
     def _is_church(self):
         """ Tell if the given Partners are Church Partners
             (by looking at their categories). """
@@ -99,9 +98,8 @@ class ResPartner(models.Model):
             if church_category in record.category_id:
                 is_church = True
 
-            record.church_member_count = len(record.member_ids.ids)
+            record.church_member_count = len(record.member_ids)
             record.is_church = is_church
-
 
     @api.multi
     def get_unreconciled_amount(self):
@@ -117,6 +115,36 @@ class ResPartner(models.Model):
         for move_line in move_line_ids:
             res += move_line.credit
         return res
+
+    @api.multi
+    def _compute_children(self):
+        for partner in self:
+            partner.number_children = len(partner.sponsored_child_ids)
+            if partner.is_church:
+                partner.number_children += len(partner.mapped(
+                    'member_ids.sponsored_child_ids'))
+
+    @api.multi
+    @api.depends('category_id')
+    def _compute_has_sponsorships(self):
+        """
+        A partner is sponsor if he is correspondent of at least one
+        sponsorship.
+        """
+        for partner in self:
+            if partner.is_church:
+                nb_sponsorships = self.env['recurring.contract'].search_count(
+                    ['|', '|',
+                        ('correspondant_id', 'in', partner.member_ids.ids),
+                        ('correspondant_id', '=', partner.id), '|',
+                        ('partner_id', '=', partner.id),
+                        ('partner_id', 'in', partner.member_ids.ids),
+                        ('type', 'like', 'S')
+                     ])
+                partner.has_sponsorships = nb_sponsorships
+                partner.number_sponsorships = nb_sponsorships
+            else:
+                super(ResPartner, self)._compute_has_sponsorships()
 
     ##########################################################################
     #                              ORM METHODS                               #
@@ -273,6 +301,45 @@ class ResPartner(models.Model):
         if record:
             partner = self.browse(record[1])
         return record and partner.lang
+
+    @api.multi
+    def open_sponsored_children(self):
+        self.ensure_one()
+        if self.is_church:
+            return {
+                'type': 'ir.actions.act_window',
+                'name': 'Children',
+                'res_model': 'compassion.child',
+                'view_type': 'form',
+                'view_mode': 'tree,form',
+                'domain': ['|', ('sponsor_id', '=', self.id),
+                           ('sponsor_id', 'in', self.member_ids.ids)],
+                'context': self.env.context,
+            }
+        else:
+            return super(ResPartner, self).open_sponsored_children()
+
+    @api.multi
+    def open_contracts(self):
+        """ Used to bypass opening a contract in popup mode from
+        res_partner view. """
+        self.ensure_one()
+        if self.is_church:
+            return {
+                'type': 'ir.actions.act_window',
+                'name': 'Contracts',
+                'res_model': 'recurring.contract',
+                'views': [[False, "tree"], [False, "form"]],
+                'domain': ['|', '|',
+                           ('correspondant_id', '=', self.member_ids.ids),
+                           ('correspondant_id', '=', self.id), '|',
+                           ('partner_id', '=', self.id),
+                           ('partner_id', 'in', self.member_ids.ids)],
+                'context': self.with_context({
+                    'default_type': 'S'}).env.context,
+            }
+        else:
+            return super(ResPartner, self).open_contracts()
 
     ##########################################################################
     #                             PRIVATE METHODS                            #

--- a/partner_compassion/views/partner_compassion_view.xml
+++ b/partner_compassion/views/partner_compassion_view.xml
@@ -96,9 +96,10 @@
 
             <!-- Add the church members list in a church partner -->
             <xpath expr="//page[2]" position="after">
-                <page string="Members" attrs="{'invisible': ['|',('is_church','=',False), ('member_ids', '=', [])]}" autofocus="autofocus">Member(
-                    <field name="church_member_count"/>)
-
+                <page string="Members" attrs="{'invisible': ['|',('is_church','=',False), ('member_ids', '=', [])]}" autofocus="autofocus">
+                    <group>
+                        <field name="church_member_count"/>
+                    </group>
                     <field name="member_ids" mode="kanban" context="{'default_church_id': active_id, 'default_city': city, 'default_state_id': state_id, 'default_zip': zip, 'default_country_id': country_id}"/>
                 </page>
             </xpath>

